dgraph/cmd/zero/zero_test.go

```diff
@@ -83,3 +89,30 @@ func TestIdBump(t *testing.T) {
 	res, err = zc.AssignIds(ctx, &pb.Num{Val: 10, Type: pb.Num_UID, Bump: true})
 	require.Contains(t, err.Error(), "Nothing to be leased")
 }
+
+func extractNodeIdFrom(proposalKey uint64) uint64 {
+	pkeybyteSlice := make([]byte, 8)
+	binary.BigEndian.PutUint64(pkeybyteSlice, proposalKey)
+
+	numSlice := make([]byte, 8)
+	// First two bytes in proposalKey are reserved for node Id.
+	copy(numSlice[6:8], pkeybyteSlice[0:2])
+
+	return binary.BigEndian.Uint64(numSlice)
+}
+
+func TestProposalKey(t *testing.T) {
+	dir, err := ioutil.TempDir("", "test_pk")
+	require.NoError(t, err)
+	defer os.RemoveAll(dir)
+
+	store := raftwal.Init(dir)
```

Comment:
do you need to call this? even when store is nil inside the node, the call to uniqueKey and initProposalKey should work

Reply:
Yes, this is required for correct initialization. The code crashes with nil store.

Reply:
You only need a node object, what code/function causes crash?

Reply:
https://github.com/dgraph-io/dgraph/blob/c04ce807f9399cc9d00159d54333d7a6a1de18fe/conn/node.go#L90

Reply:
Can we not live without calling newNode? We can just create the object ourselves and call these functions.

Reply:
Removed.